src/JasperBus.Tests/Transports/InMemory/inmemory_integration_specs.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Baseline;
+using Baseline.Dates;
+using JasperBus.Runtime;
+using JasperBus.Tests.Runtime;
+using Microsoft.DotNet.InternalAbstractions;
+using Shouldly;
+using Xunit;
+using Platform = Baseline.Platform;
+
+namespace JasperBus.Tests.Transports.InMemory
+{
+    public class inmemory_integration_specs : IntegrationContext
+    {
+        private readonly MessageTracker theTracker = new MessageTracker();
+
+        public inmemory_integration_specs()
+        {
+            with(_ =>
+            {
+                _.UseInMemoryTransport();
+                _.ListenForMessagesFrom("memory://incoming");
+                _.SendMessage<Message1>().To("memory://incoming");
+
+                _.Services.For<MessageTracker>().Use(theTracker);
+
+                _.Services.Scan(x =>
+                {
+                    x.TheCallingAssembly();
+                    x.WithDefaultConventions();
+                });
+            });
+        }
+
+
+        [Fact]
+        public async Task send_a_message_and_get_the_response()
+        {
+            var bus = Runtime.Container.GetInstance<IServiceBus>();
+
+            var task = theTracker.WaitFor<Message1>();
```

Comment:
We should probably go make the `WaitFor` a little more intelligent so that you can express timeouts as a method. Use a Task.Delay() internally to do the timeout maybe?